bigtable/filters/filters.go

```diff
@@ -0,0 +1,274 @@
+}
+
+// [END bigtable_filters_limit_row_regex]
+// [START bigtable_filters_limit_cells_per_col]
+func filterLimitCellsPerCol(w io.Writer, projectID, instanceID string, tableName string) error {
+	filter := bigtable.LatestNFilter(2)
+	return readWithFilter(w, projectID, instanceID, tableName, filter)
+}
+
+// [END bigtable_filters_limit_cells_per_col]
+// [START bigtable_filters_limit_cells_per_row]
+func filterLimitCellsPerRow(w io.Writer, projectID, instanceID string, tableName string) error {
+	filter := bigtable.CellsPerRowLimitFilter(2)
+	return readWithFilter(w, projectID, instanceID, tableName, filter)
+}
+
+// [END bigtable_filters_limit_cells_per_row]
+// [START bigtable_filters_limit_cells_per_row_offset]
+func filterLimitCellsPerRowOffset(w io.Writer, projectID, instanceID string, tableName string) error {
+	filter := bigtable.CellsPerRowOffsetFilter(2)
+	return readWithFilter(w, projectID, instanceID, tableName, filter)
+}
+
+// [END bigtable_filters_limit_cells_per_row_offset]
+// [START bigtable_filters_limit_col_family_regex]
+func filterLimitColFamilyRegex(w io.Writer, projectID, instanceID string, tableName string) error {
+	filter := bigtable.FamilyFilter("stats_.*$")
+	return readWithFilter(w, projectID, instanceID, tableName, filter)
+}
+
+// [END bigtable_filters_limit_col_family_regex]
+// [START bigtable_filters_limit_col_qualifier_regex]
+func filterLimitColQualifierRegex(w io.Writer, projectID, instanceID string, tableName string) error {
+	filter := bigtable.ColumnFilter("connected_.*$")
+	return readWithFilter(w, projectID, instanceID, tableName, filter)
+}
+
+// [END bigtable_filters_limit_col_qualifier_regex]
+// [START bigtable_filters_limit_col_range]
+func filterLimitColRange(w io.Writer, projectID, instanceID string, tableName string) error {
+	filter := bigtable.ColumnRangeFilter("cell_plan", "data_plan_01gb", "data_plan_10gb")
+	return readWithFilter(w, projectID, instanceID, tableName, filter)
+}
+
+// [END bigtable_filters_limit_col_range]
+// [START bigtable_filters_limit_value_range]
+func filterLimitValueRange(w io.Writer, projectID, instanceID string, tableName string) error {
+	filter := bigtable.ValueRangeFilter([]byte("PQ2A.190405"), []byte("PQ2A.190406"))
+	return readWithFilter(w, projectID, instanceID, tableName, filter)
+}
+
+// [END bigtable_filters_limit_value_range]
+// [START bigtable_filters_limit_value_regex]
+func filterLimitValueRegex(w io.Writer, projectID, instanceID string, tableName string) error {
+	filter := bigtable.ValueFilter("PQ2A.*$")
+	return readWithFilter(w, projectID, instanceID, tableName, filter)
+}
+
+// [END bigtable_filters_limit_value_regex]
+// [START bigtable_filters_limit_timestamp_range]
+func filterLimitTimestampRange(w io.Writer, projectID, instanceID string, tableName string) error {
+	startTime := (bigtable.Timestamp(0)).Time()
+	timestampMinusHr := bigtable.Timestamp(bigtable.Now().TruncateToMilliseconds() - 60*60*1000*1000)
+	endTime := timestampMinusHr.Time()
```

Comment:
Do these need to go back and forth between `bigtable.Timestamp` and `time.Time`? Consider

```go
endTime := time.Now()
startTime := endTime.Add(-1*time.Hour)
filter := bigtable.TimestampRangeFilter(startTime, endTime)
```

(Adjust if you want the range to be the beginning of time (`time.Unix(0, 0)`?) to one hour ago.)